prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -1061,23 +986,6 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
     ) -> SpannedEncodingResult<(Vec<vir::Stmt>, Option<MirSuccessor>)> {
         debug!("Encode location {:?}", location);
 
-        // Encode statements to inform fold-unfold of the downcasts
-        let mut downcast_stmts = vec![];
-        for (place, variant_idx) in self.mir_encoder.get_downcasts_at_location(location).into_iter() {
-            let (encoded_place, pre_stmts, place_ty, _) = self
-                    // TODO: may need to look at place to decide the arrayaccesskind here
-                    .encode_projection(place.local, &place.projection, ArrayAccessKind::Shared, location)?;
-            downcast_stmts.extend(pre_stmts);
-            let variant_field = if let ty::TyKind::Adt(adt_def, _) = place_ty.kind() {
-                let tcx = self.encoder.env().tcx();
-                let variant_name = adt_def.variants[variant_idx].ident(tcx).to_string();
-                self.encoder.encode_enum_variant_field(&variant_name)
-            } else {
-                unreachable!()
-            };
-            downcast_stmts.push(vir::Stmt::Downcast( vir::Downcast {base: encoded_place, field: variant_field} ));
-        }
-
```

Comment:
This looks suspicious to me. Without this, are fold-unfold statements for variants of enums generated correctly?